controllers/front/ProductController.php

```diff
@@ -1064,7 +1085,7 @@ public function getTemplateVarProduct()
         $product['id_product'] = (int) $this->product->id;
         $product['out_of_stock'] = (int) $this->product->out_of_stock;
         $product['new'] = (int) $this->product->new;
-        $product['id_product_attribute'] = $this->getIdProductAttributeByGroupOrRequestOrDefault();
+        $product['id_product_attribute'] = $this->getIdProductAttributeByGroupOrRequest();
```

Comment:
Be careful you are reverting the behavior that was introduced by this PR: https://github.com/PrestaShop/PrestaShop/pull/16423

Careful not to reintroduce the previously fixed bug, this attribute fetching feature is a hell I know.. It has too many sub cases and the code is poorly written (sadly we don't have much valid way of doing this better in this legacy controller context).

Besides you only changed this call in one place in this controller, whereas the initial `getIdProductAttributeByGroupOrRequestOrDefault` is called in several places, theoraticaly it should be called everywhere for consistency. But maybe that's the point of your fix? In ajax we shouldn't fallback on available product?

Reply:
I noticed that taking the available id_product_attribute made us take the wrong product quantity
https://github.com/PrestaShop/PrestaShop/blob/develop/classes/Product.php#L4978
By re-testing it right now, I see that the other attributes are changed too.

So making the "product-vaiants" container replaced well do the job. I will just keep the JS modification